src/main/java/com/okta/tools/helpers/HttpHelper.java

```diff
@@ -0,0 +1,89 @@
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.protocol.HttpContext;
+import org.apache.http.protocol.HttpCoreContext;
+import org.apache.http.ssl.SSLContexts;
+
+import javax.net.ssl.SSLContext;
+import java.io.IOException;
+import java.net.*;
+
+// Inspired by https://stackoverflow.com/a/22960881/154527
+public final class HttpHelper {
+    public static CloseableHttpClient createClient(HttpClientBuilder httpClientBuilder)
+    {
+        Registry<ConnectionSocketFactory> reg = RegistryBuilder.<ConnectionSocketFactory>create()
+                .register("http", ProxySelectorPlainConnectionSocketFactory.INSTANCE)
+                .register("https", new ProxySelectorSSLConnectionSocketFactory(SSLContexts.createSystemDefault()))
+                .build();
+        PoolingHttpClientConnectionManager cm = new PoolingHttpClientConnectionManager(reg);
+        return httpClientBuilder
+                .useSystemProperties()
+                .setConnectionManager(cm)
+                .build();
+    }
+
+    public static CloseableHttpClient createClient()
+    {
+        return createClient(HttpClients.custom());
+    }
+
+    private enum ProxySelectorPlainConnectionSocketFactory implements ConnectionSocketFactory {
+        INSTANCE;
+
+        @Override
+        public Socket createSocket(HttpContext context) throws IOException {
+            Socket socket = HttpHelper.createSocket(context);
+            return socket != null ? socket : PlainConnectionSocketFactory.INSTANCE.createSocket(context);
+        }
+
+        @Override
+        public Socket connectSocket(int connectTimeout, Socket sock, HttpHost host, InetSocketAddress remoteAddress, InetSocketAddress localAddress, HttpContext context) throws IOException {
+            return PlainConnectionSocketFactory.INSTANCE.connectSocket(connectTimeout, sock, host, remoteAddress, localAddress, context);
+        }
+    }
+
+    private static final class ProxySelectorSSLConnectionSocketFactory extends SSLConnectionSocketFactory {
+        ProxySelectorSSLConnectionSocketFactory(SSLContext sslContext) {
+            super(sslContext);
+        }
+
+        @Override
+        public Socket createSocket(HttpContext context) throws IOException {
+            Socket socket = HttpHelper.createSocket(context);
+            return socket != null ? socket : super.createSocket(context);
+        }
+    }
+
+    private static Socket createSocket(HttpContext context) {
+        HttpHost httpTargetHost = (HttpHost) context.getAttribute(HttpCoreContext.HTTP_TARGET_HOST);
+        URI uri = URI.create(httpTargetHost.toURI());
+        Proxy proxy = ProxySelector.getDefault().select(uri).iterator().next();
+        return proxy.type().equals(Proxy.Type.SOCKS) ? new Socket(proxy) : null;
```

Comment:
Create a SOCKS socket if there is a SOCKS proxy. Default to null. Java's Socket type supports HTTP proxies, but Apache HTTP Client is not compatible with it.